halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/providers/dockerRegistry/DockerRegistryCommand.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.cli.command.v1.providers.dockerRegistry;
+
+import com.beust.jcommander.Parameters;
+import com.netflix.spinnaker.halyard.cli.command.v1.NestableCommand;
+import com.netflix.spinnaker.halyard.cli.command.v1.providers.AbstractProvider;
+import com.netflix.spinnaker.halyard.cli.ui.v1.AnsiUi;
+import lombok.AccessLevel;
+import lombok.Getter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This is a top-level command for dealing with your kubernetes provider.
+ *
+ * Usage is `$ hal config`
+ */
+@Parameters()
+public class DockerRegistryCommand extends AbstractProvider {
```

Comment:
woa, i am totally confused.
When you had subcommands above in the ProviderCommand (extends NestableCommand) I thought those were commands though didnt understand where you were going with that. Here I see it is actually a provider. I'm confused but will keep reading.

I missed the part earlier where AbstractProvider is a Command so now this makes more sense. I wish AbstractProvider were called AbstractProviderCommand.